src/pages/en/guides/upgrade-to/v2.mdx

```diff
@@ -0,0 +1,359 @@
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import mdx from '@astrojs/mdx';
+
+export default defineConfig({
+  markdown: {
+    syntaxHighlight: 'prism',
+    remarkPlugins: [remarkPlugin1],
+    gfm: true,
+  },
+  integrations: [
+    mdx({
+      // `syntaxHighlight` inherited from Markdown
+
+      // Markdown `remarkPlugins` ignored,
+      // only `remarkPlugin2` applied.
+      remarkPlugins: [remarkPlugin2],
+      // `gfm` overridden to `false`
+      gfm: false,
+    })
+  ]
+});
+```
+
+##### What should I do?
+
+Revisit your Markdown and MDX configuration and compare your existing config with the new options available.
+
+:::tip[remember]
+- Astro's two default Markdown plugins are now individually configurable, and are not removed when you add other plugins. No more extending defaults!
+- The option to extend your Markdown plugin config to MDX has moved from a selection (just defaults or all plugins) to toggle (all or none).
+- Every Markdown config option (except drafts) can also be configured independently in MDX!
+:::
+
+### Added: Plugin access to frontmatter
+
+In 1.0, remark and rehype plugins did not have access to user frontmatter. Astro merged plugin frontmatter with your file's frontmatter, without passing the file frontmatter to your plugins.
+
+Astro 2.0 gives remark and rehype plugins access to user frontmatter via frontmatter injection. This allows plugin authors to modify a user's existing frontmatter, or compute new properties based on other properties.
+
+
+#### What should I do?
+
+Check any remark and rehype plugins you have written to see whether their behavior has changed. Note that `data.astro.frontmatter` is now the _complete_ Markdown or MDX document's frontmatter, rather than an empty object.
+
+### Reserved: `src/content/`
+
+In 1.0, the [Collections API was hidden behind an experimental flag](#astro-20-experimental-flags-removed). This API reserves `src/content/` as a special folder.
+
+Astro 2.0 now includes the Collections API in the core codebase for organizing your Markdown and MDX files into [content collections](/en/guides/content-collections/).
+
+#### What should I do?
+
+Rename an existing `src/content/` folder to avoid conflicts. This folder, if it exists, can now only be used for [content collections](/en/guides/content-collections/).
+
+## Astro 2.0 Experimental Flags Removed
+
+Astro v2.0 brings some new features from experimental flags into the core codebase.
+
+#### What should I do?
+
+Remove the following experimental flags from `astro.config.mjs`:
+
+```js del={5-7}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: { contentCollections: true },
+  experimental: { prerender: true },
+  experimental: { error-overlay: true },
+})
+```
```

Comment:
Given you wouldn’t have three `experimental` entries in your config, maybe it’s better to show this this way? (Also corrected the error overlay key.)

````suggestion
```js del={5-8}
// astro.config.mjs
import { defineConfig } from 'astro/config';

export default defineConfig({
  experimental: {
    contentCollections: true,
    prerender: true,
    errorOverlay: true,
  },
})
```
````